python/lsst/daf/butler/core/dimensions/standard.py

```diff
@@ -0,0 +1,407 @@
+        cached: bool,
+        viewOf: Optional[str],
+        alwaysJoin: bool,
+    ):
+        super().__init__(name)
+        self._required = required
+        self._implied = implied
+        self._topology: Dict[TopologicalSpace, StandardTopologicalFamily] = {}
+        self._metadata = metadata
+        self._cached = cached
+        self._viewOf = viewOf
+        self._alwaysJoin = alwaysJoin
+
+    @property
+    def required(self) -> NamedValueAbstractSet[Dimension]:
+        # Docstring inherited from DimensionElement.
+        return self._required
+
+    @property
+    def implied(self) -> NamedValueAbstractSet[Dimension]:
+        # Docstring inherited from DimensionElement.
+        return self._implied
+
+    @property
+    def topology(self) -> Mapping[TopologicalSpace, TopologicalFamily]:
+        # Docstring inherited from TopologicalRelationshipEndpoint
+        return MappingProxyType(self._topology)
+
+    @property
+    def metadata(self) -> NamedValueAbstractSet[ddl.FieldSpec]:
+        # Docstring inherited from DimensionElement.
+        return self._metadata
+
+    @property
+    def cached(self) -> bool:
+        # Docstring inherited from DimensionElement.
+        return self._cached
+
+    @property
+    def viewOf(self) -> Optional[str]:
+        # Docstring inherited from DimensionElement.
+        return self._viewOf
+
+    @property
+    def alwaysJoin(self) -> bool:
+        # Docstring inherited from DimensionElement.
+        return self._alwaysJoin
+
+
+class StandardDimensionElementConstructionVisitor(DimensionConstructionVisitor):
+    """A construction visitor for `StandardDimension` and
+    `StandardDimensionCombination`.
+
+    Parameters
+    ----------
+    name : `str`
+        Name of the dimension.
+    required : `Set` [ `Dimension` ]
+        Names of dimensions whose keys define the compound primary key for this
+        element's (logical) table, as well as references to there own
+        tables.
+    implied : `Set` [ `Dimension` ]
+        Names of dimension whose keys are included in this elements's
+        (logical) table as foreign keys.
+    metadata : `Iterable` [ `ddl.FieldSpec` ]
+        Field specifications for all non-key fields in this element's table.
+    cached : `bool`
+        Whether to cache the records of this element in memory when they
+        are fetched from the database.
+    viewOf : `str`, optional
+        Name of another `StandardDimension` or `StandardDimensionCombination`
+        whose records this element's records summarize.
+    uniqueKeys : `Iterable` [ `ddl.FieldSpec` ]
+        Fields that can each be used to uniquely identify this dimension (given
+        values for all required dimensions).  The first of these is used as
+        (part of) this dimension's table's primary key, which others used to
+        define unique constraints.  Should be empty for
+        `StandardDimensionCombination` definitions.
+    """
```

Comment:
`alwaysJoin` missing here too

Reply:
Lot's of copy-paste for parameter descriptions, would be nice if sphinx could automate that.

Reply:
Yeah, I think copy-paste here is just the lesser of two evils.  It's probably a fundamental Python limitation, not a Sphinx one, at least as long as we care about having docstrings as well as HTML docs.